### ISSUE TITLE
Failing tests in local brew test env

### ISSUE THREAD

**Issue by MatzFan** [Homebrew/brew#1107]
Am seeing failing tests in my local test env setup with `brew --env=std`. Here is a [gist](https://gist.github.com/MatzFan/0edffa8bcd2a48c79dbb313aba9a8815) showing my local setup and the result of `brew tests`.

Be grateful for help in getting me green so I can contribute :)

**Edit**
```
brew ~/dev/Homebrew/brew$ brew doctor
Your system is ready to brew.
```

**Comment by zmwangx** [comment]
Those happen all the time.

**Comment by zmwangx** [comment]
I usually just run `brew tests --only=<what-i-changed>`.

**Comment by MatzFan** [comment]
Reassuring, will do that.

**Comment by MikeMcQuaid** [comment]
Definitely shouldn't be failing all the time.

**Comment by MatzFan** [comment]
A consistently failing integration test masked a failing assertion I'd added in the same test in #1088 - hence the bot failed it in CI. Be nice to start with a clean slate.

**Comment by MatzFan** [comment]
Seems to be a more fundamental problem; whatever test files are being run, they are not the ones on my local master branch. In this [gist](https://gist.github.com/MatzFan/49005163e31a8832643f43391d8e07c3) I ran `test_resource.rb` only, as this highlights one of the failing tests, then _removed_ that file, ran it again - same result!! Maybe my git is corrupt - anyone seen anything like this before?

**Comment by MikeMcQuaid** [comment]
Which test is consistently failing for you both?

**Comment by MatzFan** [comment]
@MikeMcQuaid for me these 5, still with `master` right up to date this morning - but see my comment above re odd behaviour:

- `ResourceTests#test_verify_download_integrity_mismatch`
- `AbstractDownloadStrategyTests#test_expand_safe_system_args_with_explicit_quiet_flag`
- `AbstractDownloadStrategyTests#test_expand_safe_system_args_with_implicit_quiet_flag`
- `IntegrationCommandTestTap#test_tap`
- `IntegrationCommandTestOutdated#test_outdated`

**Comment by MikeMcQuaid** [comment]
@MatzFan Could you try to fix them and submit fixes? Hard to do so myself when I and the bot can't reproduce 😭

**Comment by MatzFan** [comment]
Will do

**Comment by MatzFan** [comment]
So I can fix the tests, but before I do I need at least a basic understanding what the `brew sh` command is actually doing. I thought `brew tests` would run tests in my fork of the brew repo (cloned into `~/dev/brew`), but I see that the command - when run from inside the brew test env shell - ignores the local git repo and actually runs tests on the files in my Homebrew installation (`/usr/local/Homebrew`).

Is this correct and if so how do I work on these files without origin pointing to my fork 😕 ?? Sorry for contributor noob question, be fine once I have got this.

**Comment by MatzFan** [comment]
Fixed by #1193
